mutiny-core/src/federation.rs

```diff
@@ -0,0 +1,612 @@
+                let logger = self.logger.clone();
+                let lightning_module = Arc::clone(&lightning_module);
+                async move {
+                    if entry.operation_module_kind() == "ln" {
+                        Some(
+                            handle_ln_meta(logger, &entry, key.operation_id, &lightning_module)
+                                .await,
+                        )
+                    } else {
+                        log_warn!(
+                            logger,
+                            "Unsupported module: {}",
+                            entry.operation_module_kind()
+                        );
+                        None
+                    }
+                }
+            })
+            .filter_map(|item| async move { item })
+            .collect::<Vec<ActivityItem>>()
+            .await;
+
+        Ok(activity_items)
+    }
+
+    pub async fn get_invoice_by_hash(
+        &self,
+        hash: &sha256::Hash,
+    ) -> Result<MutinyInvoice, MutinyError> {
+        let operations = self
+            .fedimint_client
+            .operation_log()
+            .list_operations(100, None)
+            .await;
+
+        let lightning_module = self
+            .fedimint_client
+            .get_first_module::<LightningClientModule>();
+
+        for (key, entry) in operations {
+            if entry.operation_module_kind() == "ln" {
+                if let Some(invoice) = extract_invoice_from_entry(
+                    self.logger.clone(),
+                    &entry,
+                    hash,
+                    key.operation_id,
+                    &lightning_module,
+                )
+                .await
+                {
+                    return Ok(invoice);
+                }
+            } else {
+                log_warn!(
+                    self.logger,
+                    "Unsupported module: {}",
+                    entry.operation_module_kind()
+                );
+            }
+        }
+
+        Err(MutinyError::NotFound)
+    }
+
+    pub(crate) async fn pay_invoice(
+        &self,
+        invoice: &Bolt11Invoice,
+    ) -> Result<MutinyInvoice, MutinyError> {
+        let lightning_module = self
+            .fedimint_client
+            .get_first_module::<LightningClientModule>();
+        let outgoing_payment = lightning_module
+            .pay_bolt11_invoice(invoice.clone(), ())
+            .await?;
+
+        // Subscribe and process outcome based on payment type
+        let inv = match outgoing_payment.payment_type {
+            fedimint_ln_client::PayType::Internal(pay_id) => {
+                match lightning_module.subscribe_internal_pay(pay_id).await {
+                    Ok(o) => {
+                        process_outcome(
+                            o,
+                            process_pay_state_internal,
+                            invoice.clone(),
+                            true,
+                            30_000,
```

Comment:
should just pull in `DEFAULT_PAYMENT_TIMEOUT * 1_000` from `node.rs`  instead of this magic value